.github/workflows/merge-main-into-staging.yml

```diff
@@ -0,0 +1,21 @@
+name: Merge main into staging
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  merge:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Merge main into staging
+        run: |
+          git config user.name "GitHub Actions"
+          git config user.email "actions@github.com"
+          git checkout staging
+          git pull
+          git merge --no-ff main -m "Merge main into staging"
```

Comment:
Is the `--no-ff` strictly necessary? Just curious.

Reply:
_With the `--no-ff` option on it creates a merge commit in all cases_. This means every merge into `staging` will be merge commit even if the current HEAD of `staging` is an ancestor of the incoming commit from `main`. Without the `--no-ff` you won't see a merge commit on `staging` for the above case - instead the branch pointer would be moved up to point to the commit at the HEAD of `main`.

From the perspective of someone looking at the `staging` branch, seeing a merge commit like `Merge ...` more easily allows them to see that the GHA executed and what was added from `main` when this GHA executes.

I could remove it if you feel strongly that explicit merge commits are not helpful and are too noisy

Reply:
i might argue against it since we will be deploying ~continuously~ 😄

Reply:
Agreed--since _every commit_ to main is going to be merged to staging, it's probably not adding much to see the merge commit.

Reply:
will remove